### PR TITLE
Add terraform deploying infrastructure for benchmarks

### PR DESCRIPTION
**Pull request by grzegorz8**
## Description
Currently, in order to run performance benchmarks one need to create the infrastructure manually. This PR adds Terraform scripts which do that automatically for AWS and GCP.

## How was this patch tested?
I tested this patch manually on AWS and GCP cloud.

## Does this PR introduce _any_ user-facing changes?
No.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -114,3 +114,19 @@ pycodestyle*.py
 
 # For IDE settings
 .vscode
+
+# For Terraform
+**/.terraform/*
+*.tfstate
+*.tfstate.*
+*.terraform.lock*
```
Comment (alexott) [inline .gitignore:122]: it's better to not ignore `.terraform.lock.hcl`: https://www.terraform.io/language/files/dependency-lock#lock-file-location

**Comment by dennyglee** [comment]
@vinijaiswal

File: benchmarks/infrastructure/aws/terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+   availability_zone1     = "<AVAILABILITY_ZONE1>"
+   availability_zone2     = "<AVAILABILITY_ZONE2>"
+   benchmarks_bucket_name = "<BUCKET_NAME>"
+   source_bucket_name     = "<SOURCE_BUCKET_NAME>"
+   mysql_user             = "<MYSQL_USER>"
+   mysql_password         = "<MYSQL_PASSWORD>"
+   emr_public_key_path    = "<EMR_PUBLIC_KEY_PATH>"
+   user_ip_address        = "<MY_IP>"
+   emr_workers            = WORKERS_COUNT
+   emr_public_key_path    = "<PUBLIC_KEY_PATH>"
```
Comment (rdziadosz) [inline benchmarks/infrastructure/aws/terraform/README.md:28]: Duplicate

File: benchmarks/infrastructure/gcp/terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Create infrastructure with Terraform
+
+1. Install [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started).
+
+2. Create and download [service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-console).
+   The path to the file is required in the next step (`credentials_file`).
+
```
Comment (rdziadosz) [inline benchmarks/infrastructure/gcp/terraform/README.md:7]: I would add:
Make sure the service account has the [permissions](https://console.cloud.google.com/iam-admin/iam) listed in prerequisites.

File: benchmarks/infrastructure/gcp/terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Create infrastructure with Terraform
+
+1. Install [Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started).
+
+2. Create and download [service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-console).
```
Comment (rdziadosz) [inline benchmarks/infrastructure/gcp/terraform/README.md:5]: I would add: key file should be in JSON format

File: benchmarks/infrastructure/aws/terraform/README.md (hunk 2 above)
Comment (grzegorz8) [reply benchmarks/infrastructure/aws/terraform/README.md:28]: removed

File: benchmarks/infrastructure/gcp/terraform/README.md (hunk 3 above)
Comment (grzegorz8) [reply benchmarks/infrastructure/gcp/terraform/README.md:7]: done

File: benchmarks/infrastructure/gcp/terraform/README.md (hunk 4 above)
Comment (grzegorz8) [reply benchmarks/infrastructure/gcp/terraform/README.md:5]: done

File: .gitignore (hunk 1 above)
Comment (grzegorz8) [reply .gitignore]: Thanks, I removed this line and I added `.terraform.lock.hcl` files to the repository.

**Comment by dennyglee** [comment]
LGTM!

**Review by dennyglee — APPROVED** [review]

File: benchmarks/README.md [hunk 5]
```diff
@@ -64,10 +68,14 @@ _________________
   or assigned to the [master node](https://cloud.google.com/compute/docs/connect/add-ssh-keys#after-vm-creation) only.
 - Ideally, all GCP components used in benchmark should be in the same location (Storage bucket, Dataproc Metastore service and Dataproc cluster).
 
-#### Prepare GCS bucket
+There are two ways to create infrastructure required for benchmarks - using provided [Terraform template](infrastructure/gcp/terraform/README.md) or manually.
+
+#### Create infrastructure manually
```
Comment (tdas) [inline benchmarks/README.md:73]: Since using terraform is not separation section, I dont think this needs to be a section on its own.

Comment (grzegorz8) [reply benchmarks/README.md]: Ok, I removed the header (both in line 26 and 73).

**Review by tdas — APPROVED** [review]

**Review by tdas — APPROVED** [review]

**Pull request closed without merging**